### PR TITLE
CCCT-567 || Home Screen UI Fixes

### PR DESCRIPTION
**Pull request by j13panchal**
## Summary
https://dimagi.atlassian.net/browse/CCCT-567

## PR Checklist

- [ ] If I think the PR is high risk, "High Risk" label is set
- [ ] I have confidence that this PR will not introduce a regression for the reasons below
- [ ] Do we need to enhance manual QA test coverage ? If yes, "QA Note" label is set correctly
- [ ] Does the PR introduce any major changes worth communicating ? If yes, "Release Note" label is set and a "Release Note" is specified in PR description.

### CODE REVIEW

File: app/src/org/commcare/activities/HomeButtons.java [hunk 1]
```diff
@@ -27,8 +29,8 @@
  */
 public class HomeButtons {
 
-    private final static String[] buttonNames =
```
Comment (OrangeAndGreen) [inline app/src/org/commcare/activities/HomeButtons.java:30]: Apologies, I should have provided more guidance here. Hiding the button can be accomplished more easily and without having to make additional modifications to this code. Instead, we just need to enhance StandardHomeActivityUIController.getHiddenButtons so it correctly indicates when to hide the Job Status button. The list of button names returned by that function is passed through to HomeButtons so buttons can be shown or hidden as indicated.

File: app/src/org/commcare/connect/ConnectManager.java [hunk 2]
```diff
@@ -985,4 +986,11 @@ public static String generatePassword() {
 
         return password.toString();
     }
+
+    public static boolean shouldShowJobStatus(Activity activity) {
```
Comment (OrangeAndGreen) [inline app/src/org/commcare/connect/ConnectManager.java:990]: StandardHomeActivityUIController.getHiddenButtons an call this method to determine whether to add "connect" to the list of hidden buttons.

File: app/src/org/commcare/connect/ConnectManager.java [hunk 3]
```diff
@@ -985,4 +986,11 @@ public static String generatePassword() {
...
+        String appId = CommCareApplication.instance().getCurrentApp().getUniqueId();
+        ConnectAppRecord record = ConnectManager.getAppRecord(activity, appId);
+        ConnectJobRecord job = ConnectManager.getActiveJob();
+        return job.getStatus() == STATUS_DELIVERING && record.getIsLearning();
```
Comment (OrangeAndGreen) [inline app/src/org/commcare/connect/ConnectManager.java:994]: For the logic here, we want to check for two possibilities where we want to show the Job Status button:
1. Job in learning state, logged in to learning app
2. Job in delivery state, logged in to delivery app

So (status == LEARNING && record.getIsLearning) || (status == DELIVERY && !record.getisLearning)

**Review by OrangeAndGreen — CHANGES_REQUESTED** [review]

**Comment by shubham1g5** [comment]
(Testing code rabbit here)
@coderabbitai full review please

**Review by OrangeAndGreen — APPROVED** [review]

**Pull request merged**